pkg/provider/aws/resources/eks.go

```diff
@@ -326,6 +342,25 @@ func (cluster *EksCluster) GetOutputFiles() []core.File {
 	return cluster.Manifests
 }
 
+func (cluster *EksCluster) installNvidiaDevicePlugin(references []core.AnnotationKey, dag *core.ResourceGraph) {
+	manifest := &kubernetes.Manifest{
+		Name:          fmt.Sprintf("%s-%s", cluster.Name, "nvidia-device-plugin"),
+		ConstructRefs: references,
+		FilePath:      "https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v1.10/nvidia-device-plugin.yml",
+		ClustersProvider: core.IaCValue{
+			Resource: cluster,
+			Property: CLUSTER_PROVIDER_IAC_VALUE,
+		},
+	}
+
+	dag.AddResource(manifest)
+	dag.AddDependenciesReflect(manifest)
+
+	for _, ng := range cluster.getClustersNodeGroups(dag) {
+		dag.AddDependency(manifest, ng)
+	}
+}
```

Comment:
the references here may not be correct because were passing in the clusters references and not the node groups which caused this creation. We ideally want the ref to just be exec units that need this. Not blocking for now, but better for tracking